### PR TITLE
off chain verification script for ocr2vrf

### PR DESCRIPTION
**Pull request by jinhoonbang**
- off chain verification script for ocr2vrf

### CODE REVIEW

File: core/scripts/ocr2vrf/util.go [hunk 1]
```diff
@@ -520,3 +532,167 @@ func requestRandomnessCallbackBatch(
 
 	return requestID
 }
+
+func getDKGLatestConfigDetails(e helpers.Environment, dkgAddress string) dkgContract.LatestConfigDetails {
```
Comment (makramkd) [inline core/scripts/ocr2vrf/util.go:536]: Organization nit/ this file is becoming quite large, what do you think about moving all this verification-related code to a `verify.go` in the same directory?

File: core/scripts/ocr2vrf/util.go [hunk 2]
```diff
@@ -520,3 +532,167 @@ func requestRandomnessCallbackBatch(
...
+
+func getVRFSignature(e helpers.Environment, beaconAddress string, height, confDelay uint64) (proofG1X, proofG1Y *big.Int) {
+	// get transmission logs from requested block to latest block
+	// TODO: index transmission logs by height and confirmation delay to
+	// make the FilterQuery call more efficient
+	latestBlock, err := e.Ec.HeaderByNumber(context.Background(), nil)
+	helpers.PanicErr(err)
+	query := ethereum.FilterQuery{
+		FromBlock: big.NewInt(0).SetUint64(height),
+		ToBlock:   latestBlock.Number,
```
Comment (makramkd) [inline core/scripts/ocr2vrf/util.go:597]: In my testing I set this to `height + 200` or something, we usually get a response quite quickly. Narrows down the search space by a lot especially if it's an older proof.

File: core/scripts/ocr2vrf/util.go [hunk 3]
```diff
@@ -520,3 +532,167 @@ func requestRandomnessCallbackBatch(
...
+
+	beacon := newVRFBeacon(common.HexToAddress(beaconAddress), e.Ec)
+	for _, log := range logs {
+		unpacked, err := beacon.ParseLog(log)
+		helpers.PanicErr(err)
+
+		t, ok := unpacked.(*vrf_beacon.VRFBeaconNewTransmission)
+		if !ok {
+			helpers.PanicErr(errors.New("failed to convert log to VRFBeaconNewTransmission"))
+		}
```
Comment (makramkd) [inline core/scripts/ocr2vrf/util.go:617]: I think you can call `beacon.ParseNewTransmission` directly, it's clearer IMO.

**Review by makramkd — COMMENTED** [review]
Very cool!

File: core/scripts/ocr2vrf/util.go (hunk 2 above)
Comment (jinhoonbang) [reply core/scripts/ocr2vrf/util.go]: makes sense. Added a new CLI parameter for `searchWindow` and set default to 200

File: core/scripts/ocr2vrf/util.go (hunk 1 above)
Comment (jinhoonbang) [reply core/scripts/ocr2vrf/util.go]: sounds good :+1:

**Review by makramkd — APPROVED** [review]

**Pull request merged**